Dynatrace/dynatrace-operator
# Add rbac turn-off option in Helm values.yaml

[K8S-10824](https://dt-rnd.atlassian.net/browse/K8S-10824)

The following options have been added to the helm values.yaml:
```
rbac:
  activeGate:
    create: true
  edgeConnect:
    create: true
  extensions:
    create: true
  logModule:
    create: true
  oneAgent:
    create: true
```

Turning any of them to `false` will result in the following:
1. The related RBAC components (`Role`/`ClusterRole`/`ServiceAccount`) will not be deployed
2. The validation webhook will deny any `DynaKube`/`EdgeConnect` CR that tries to configure a feature relying on such RBAC components
   - This is done by creating a `ConfigMap` called `operator-config`, the content's of which is added to the envs of the webhook
     - This env is ONLY checked 1 time at startup

## How can this be tested?

Configure the `rbac.<module>.create` fields to your heart's content.
- Check the `install-config` `ConfigMap` reflect what you configred
- Try to deploy a CR that would use the disabled "module"
